02-microarray/dimension-reduction_microarray_02_umap.Rmd

```diff
@@ -0,0 +1,321 @@
+all.equal(colnames(df), metadata$geo_accession)
+```
+
+
+Now we are going to use a combination of functions from the `umap` and the `ggplot2` packages to perform and visualize the results of the Uniform Manifold Approximation (UMAP) dimension reduction technique on our medulloblastoma samples.
+
+## Perform Uniform Manifold Approximation (UMAP)
+
+In this code chunk, we are going to perform Uniform Manifold Approximation (UMAP) on our data and create a data frame using the UMAP scores and the variables from our metadata that we are going to use to annotate our plot later.
+
+```{r}
+# Perform Uniform Manifold Approximation (UMAP)
+umap_data <- umap::umap(t(df)) # We have to transpose our data frame to meet the `umap::umap()` function requirements
+
+# Make into data frame for plotting with `ggplot2`
+umap_df <- data.frame(umap_data$layout) %>%
+  # Turn samples_ids stored as rownames into column
+  tibble::rownames_to_column("refinebio_accession_code") %>%
+  # Bring only the variables that we want from the metadata into this data frame; match by sample ids
+  dplyr::inner_join(dplyr::select(
+    metadata, refinebio_accession_code,
+    histology, subgroup
+  ),
+  by = "refinebio_accession_code"
+  )
+```
+
+Now, let's plot the UMAP scores.
+
+```{r}
+# Make a scatterplot using `ggplot2` functionality
+umap_plot <- ggplot(
+  umap_df,
+  aes(
+    x = X1,
+    y = X2
+  )
+) +
+  geom_point() + # This tells R that we want a scatterplot
+  theme_classic() # This tells R to return a classic-looking plot with no gridlines
+
+# Print out plot here
+umap_plot
+```
+
+Now, let's label the data points based on their genotype subgroup since this is central to the subgroup specific based hypothesis in the [original paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3683624/#SD1) [@Northcott2012].
+
+```{r}
+# Make a scatterplot with ggplot2
+umap_plot <- ggplot(
+  umap_df,
+  aes(
+    x = X1,
+    y = X2,
+    color = subgroup # This will label points with different colors for each `subgroup`
+  )
+) +
+  geom_point() +
+  theme_classic()
+
+# Print out plot here
+umap_plot
+```
+
+We can add another label to our plot.
+Let's also label the data points based on the histological subtype that each sample belongs to.
+
+```{r}
+# Make a scatterplot with ggplot2
+umap_plot <- ggplot(
+  umap_df,
+  aes(
+    x = X1,
+    y = X2,
+    color = subgroup, # This will label points with different colors for each `subgroup`
+    shape = histology # This will label points with different colors for each `histology` group
+  )
+) +
+  geom_point() +
+  theme_classic()
+
+# Print out plot here
+umap_plot
+```
+
+## Save annotated UMAP plot as a PNG
+
+You can easily switch this to save to a JPEG or TIFF by changing the file name within the `ggsave()` function to the respective file suffix.
+
+```{r}
+# Save plot using `ggsave()` function
+ggsave(file.path(
+  plots_dir,
+  "GSE37382_umap_scatterplot.png"
+), # Replace with name relevant your plotted data
```

Comment:
I suppose this comment should be moved up one. (I suspect `styler` might have moved this paranethesis down one line).